soh/src/code/z_parameter.c

```diff
@@ -1505,85 +1503,173 @@ void Inventory_SwapAgeEquipment(void) {
-            if (!(CHECK_OWNED_EQUIP(EQUIP_TYPE_SWORD, EQUIP_INV_SWORD_KOKIRI) == 0) && !Flags_GetInfTable(INFTABLE_SWORDLESS)) {
-                gSaveContext.equips.equipment |= EQUIP_VALUE_SWORD_KOKIRI << (EQUIP_TYPE_SWORD * 4);
+            gSaveContext.equips.equipment |= EQUIP_VALUE_SWORD_KOKIRI << (EQUIP_TYPE_SWORD * 4);
+        }
+        // In Rando we need an extra case to handle starting as adult. We can use the fact that the childEquips will be uninitialised (i.e. 0) at this point
+        else if (gSaveContext.childEquips.equipment == 0) {
+
+            //zero out items
+            for (i = 0; i < ARRAY_COUNT(gSaveContext.equips.buttonItems); i++) {
+                gSaveContext.equips.buttonItems[i] = ITEM_NONE;
+                if (i != 0) {
+                    gSaveContext.equips.cButtonSlots[i-1] = ITEM_NONE;
+                }
             }
-        } else if (gSaveContext.childEquips.buttonItems[0] != ITEM_NONE) {
+            gSaveContext.equips.equipment = (EQUIP_VALUE_SWORD_KOKIRI << (EQUIP_TYPE_SWORD * 4)) |
+                                            (EQUIP_VALUE_SHIELD_DEKU << (EQUIP_TYPE_SHIELD * 4)) |
+                                            (EQUIP_VALUE_TUNIC_KOKIRI << (EQUIP_TYPE_TUNIC * 4)) |
+                                            (EQUIP_VALUE_BOOTS_KOKIRI << (EQUIP_TYPE_BOOTS * 4));
+        }
+
+        // When becoming child in rando, set swordless flag and clear B button if player doesn't have kokiri sword
+        // Otherwise, equip sword and unset flag
+        if (!CHECK_OWNED_EQUIP(EQUIP_TYPE_SWORD, EQUIP_INV_SWORD_KOKIRI)) {
+            gSaveContext.equips.equipment &= (u16) ~(0xF << (EQUIP_TYPE_SWORD * 4));
+            gSaveContext.equips.buttonItems[0] = ITEM_NONE;
+            Flags_SetInfTable(INFTABLE_SWORDLESS);
+        } else {
+            gSaveContext.equips.equipment &= (u16) ~(0xF << (EQUIP_TYPE_SWORD * 4));
+            gSaveContext.equips.equipment |= (EQUIP_VALUE_SWORD_KOKIRI << (EQUIP_TYPE_SWORD * 4));
+            gSaveContext.equips.buttonItems[0] = ITEM_SWORD_KOKIRI;
+            Flags_UnsetInfTable(INFTABLE_SWORDLESS);
+        }
+    }
+
+    shieldEquipValue = gEquipMasks[EQUIP_TYPE_SHIELD] & gSaveContext.equips.equipment;
+    if (shieldEquipValue) {
+        shieldEquipValue >>= gEquipShifts[EQUIP_TYPE_SHIELD];
+        if (!CHECK_OWNED_EQUIP_ALT(EQUIP_TYPE_SHIELD, shieldEquipValue - 1)) {
+            gSaveContext.equips.equipment &= gEquipNegMasks[EQUIP_TYPE_SHIELD];
+        }
+    }
+}
+
+void Inventory_SwapAgeEquipment(void) {
+    s16 i;
+    u16 shieldEquipValue;
+
+    if (IS_RANDO) {
+        Rando_Inventory_SwapAgeEquipment();
+        return;
+    } else if (CVarGetInteger("gSwitchAge", 0) || CVarGetInteger("gSwitchTimeline", 0)) {
+        // Can Utilise the Rando Flow Path
+        Rando_Inventory_SwapAgeEquipment();
+        CVarSetInteger("gSwitchTimeline", 0);
+        return;
+    }
```

Comment:
is there ever a time when switching age in rando that we'd want to keep `gSwitchTimeline` set to something other than zero? if not, it seems we could just have

```suggestion
    if (IS_RANDO || CVarGetInteger("gSwitchAge", 0) || CVarGetInteger("gSwitchTimeline", 0)) {
        Rando_Inventory_SwapAgeEquipment();
        return;
    }
```

and move
```cpp
CVarSetInteger("gSwitchTimeline", 0);
```
into `Rando_Inventory_SwapAgeEquipment`

Reply:
Oh yea that makes sense, I just was originally planning on writing two functions, but realised it could just be the same!

Reply:
Do you think it's preferable to move it inside Rando_Inventory_SwapAgeEquipment? Because I thought it makes it more clear if the CVar get and set are near each other

Reply:
my mindset on this stuff is very much in "how can we migrate this to VB" mode at the moment, so I was thinking it would turn into

```c
if (!GameInteractor_Should(GI_VB_SWAP_AGE_EQUIPMENT, true, NULL) {
    return;
}
```

and then in the GI should handling we'd do the rest.

i have zero issue with it just being
```suggestion
    if (IS_RANDO || CVarGetInteger("gSwitchAge", 0) || CVarGetInteger("gSwitchTimeline", 0)) {
        Rando_Inventory_SwapAgeEquipment();
        CVarSetInteger("gSwitchTimeline", 0);
        return;
    }
```

for now

Reply:
yea i had similar thoughts for the hooks, if you have no issues then I'll do it that way